Reformat old string formats to f-strings

EwoutH opened the pull request:
Reformats all the text from the old "%-formatted" and .format(...) format to the newer f-string format, as defined in [PEP 498](https://www.python.org/dev/peps/pep-0498/). This requires Python 3.6+.

[Flynt](https://github.com/ikamensh/flynt) 0.69 was used to reformat the strings. 199 f-strings were created in 72 files.

F-strings are in general more readable, concise and performant. See also: [PEP 498#Rationale](https://www.python.org/dev/peps/pep-0498/#rationale)

```
Running flynt v.0.69

Flynt run has finished. Stats:

Execution time:                            4.080s
Files modified:                            75
Character count reduction:                 1322 (0.06%)

Per expression type:
Old style (`%`) expressions attempted:     211/235 (89.8%)
`.format(...)` calls attempted:            16/20 (80.0%)
No concatenations attempted.
F-string expressions created:              199
```

tconkling commented on lib/streamlit/report_session.py:
```diff
@@ -544,7 +544,7 @@ def _enqueue_script_request(self, request, data=None):
         """
         if self._state == ReportSessionState.SHUTDOWN_REQUESTED:
-            LOGGER.warning("Discarding %s request after shutdown" % request)
+            LOGGER.warning(f"Discarding {request} request after shutdown")
```
Not your fault, but this line should be `LOGGER.warning("Discarding %s request after shutdown", request)` (it should not have used the format operator in the first place)

tconkling commented on lib/streamlit/report_thread.py:
```diff
@@ -252,7 +252,7 @@ def get_report_ctx() -> Optional[ReportContext]:
         # script "bare", and doesn't need to be warned about streamlit
         # bits that are irrelevant when not connected to a report.
-        LOGGER.warning("Thread '%s': missing ReportContext" % thread.name)
+        LOGGER.warning(f"Thread '{thread.name}': missing ReportContext")
```
Same issue - should be `LOGGER.warning("Thread '%s': missing ReportContext", thread.name)`

tconkling commented on lib/streamlit/script_runner.py:
```diff
@@ -301,7 +301,7 @@ def _run_script(self, rerun_data):
         except BaseException as e:
             # We got a compile error. Send an error event and bail immediately.
-            LOGGER.debug("Fatal script error: %s" % e)
+            LOGGER.debug(f"Fatal script error: {e}")
```
`LOGGER.debug("Fatal script error: %s", e)`

tconkling commented on lib/streamlit/server/routes.py:
```diff
@@ -116,7 +116,7 @@ def validate_absolute_path(self, root, absolute_path):
             in_memory_file_manager.get(absolute_path)
         except KeyError:
-            LOGGER.error("InMemoryFileManager: Missing file %s" % absolute_path)
+            LOGGER.error(f"InMemoryFileManager: Missing file {absolute_path}")
```
`LOGGER.error("InMemoryFileManager: Missing file %s", absolute_path)`

tconkling commented on lib/streamlit/server/routes.py:
```diff
@@ -142,18 +142,17 @@ def get_absolute_path(cls, root, path):
     @classmethod
     def get_content(cls, abspath, start=None, end=None):
-        LOGGER.debug("MediaFileHandler: GET %s" % abspath)
+        LOGGER.debug(f"MediaFileHandler: GET {abspath}")
```
`LOGGER.debug("MediaFileHandler: GET %s", abspath)`

tconkling commented on lib/streamlit/server/routes.py:
```diff
@@ -142,18 +142,17 @@ def get_absolute_path(cls, root, path):
             in_memory_file = in_memory_file_manager.get(abspath)
         except:
-            LOGGER.error("InMemoryFileManager: Missing file %s" % abspath)
+            LOGGER.error(f"InMemoryFileManager: Missing file {abspath}")
```
`LOGGER.error("InMemoryFileManager: Missing file %s", abspath)`

tconkling commented on lib/streamlit/server/routes.py:
```diff
@@ -142,18 +142,17 @@ def get_absolute_path(cls, root, path):
 
         LOGGER.debug(
-            "InMemoryFileManager: Sending %s file %s"
-            % (in_memory_file.mimetype, abspath)
+            f"InMemoryFileManager: Sending {in_memory_file.mimetype} file {abspath}"
```
```
LOGGER.debug(
    "InMemoryFileManager: Sending %s file %s",
    in_memory_file.mimetype, abspath
)
```

tconkling commented on lib/streamlit/server/routes.py:
```diff
@@ -278,7 +277,7 @@ def get(self):
             raise tornado.web.Finish()
 
-        LOGGER.debug("MessageCache HIT [hash=%s]" % msg_hash)
+        LOGGER.debug(f"MessageCache HIT [hash={msg_hash}]")
```
`LOGGER.debug("MessageCache HIT [hash=%s]", msg_hash)`

tconkling commented on lib/streamlit/server/server.py:
```diff
@@ -438,7 +438,7 @@ def _create_app(self) -> tornado.web.Application:
 
     def _set_state(self, new_state: State) -> None:
-        LOGGER.debug("Server state: %s -> %s" % (self._state, new_state))
+        LOGGER.debug(f"Server state: {self._state} -> {new_state}")
```
`LOGGER.debug("Server state: %s -> %s", self._state, new_state)`

tconkling commented on lib/streamlit/server/server.py:
```diff
@@ -606,13 +606,13 @@ def _send_message(self, session_info: SessionInfo, msg: ForwardMsg) -> None:
                 # This session has probably cached this message. Send
                 # a reference instead.
-                LOGGER.debug("Sending cached message ref (hash=%s)" % msg.hash)
+                LOGGER.debug(f"Sending cached message ref (hash={msg.hash})")
```
`LOGGER.debug("Sending cached message ref (hash=%s)", msg.hash)`

tconkling commented on lib/streamlit/server/server.py:
```diff
@@ -606,13 +606,13 @@ def _send_message(self, session_info: SessionInfo, msg: ForwardMsg) -> None:
             # If the message is already cached, this will reset its
             # age.
-            LOGGER.debug("Caching message (hash=%s)" % msg.hash)
+            LOGGER.debug(f"Caching message (hash={msg.hash})")
```
`LOGGER.debug("Caching message (hash=%s)", msg.hash)`

tconkling commented on lib/streamlit/uploaded_file_manager.py:
```diff
@@ -213,7 +213,7 @@ def remove_orphaned_files(
 
         if num_removed > 0:
-            LOGGER.debug("Removed %s orphaned files" % num_removed)
+            LOGGER.debug(f"Removed {num_removed} orphaned files")
```
`LOGGER.debug("Removed %s orphaned files", num_removed)`

tconkling reviewed (changes requested):
Thanks so much for this code hygiene pass!

The auto-formatter turned up a few instances of the format operator being used inside log statements (that is, `LOGGER.info("hello, %s" % world")` instead of `LOGGER.info("hello, %s", world)`). 

You did not introduce these issues, but it would be great to address them. I've marked each instance (I think!) in review comments. Can you make those changes? (Alternately, I can make a new PR based on this one, and make the changes myself - happy with whatever works.)

Thanks again!

EwoutH committed:
Reformat old string formats to f-strings

Reformats all the text from the old "%-formatted" and .format(...) format to the newer f-string format, as defined in PEP 498. This requires Python 3.6+.

Flynt 0.69 was used to reformat the strings. 199 f-strings were created in 72 files.

F-strings are in general more readable, concise and performant. See also: https://www.python.org/dev/peps/pep-0498/#rationale

EwoutH committed:
Fix format operator inside log statements

This commit fixed the following issue: "The auto-formatter turned up a few instances of the format operator being used inside log statements (that is, LOGGER.info("hello, %s" % world") instead of LOGGER.info("hello, %s", world))."

Co-Authored-By: Tim Conkling <tim@timconkling.com>

EwoutH commented:
@tconkling Thanks for the review! I incorporated your changes in a second commit.

PS: For a next time, you can also directly suggest changes to code with GitHub when reviewing a PR. See [Suggesting Changes on GitHub](http://haacked.com/archive/2019/06/03/suggested-changes/) and the [GitHub docs](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/commenting-on-a-pull-request) (step 6).

tconkling commented:
Thanks, and looks good! To get this merged, we just need to get all the circleci tests passing.
- To fix the formatting errors, you can run `make pyformat`. (If pyformat ends up finding unrelated formatting issues, please only commit changes to the files you touched, in case there are unrelated files.)
- All the cypress tests are failing (!) which hopefully has an easy solution. I'd suggest updating your local `develop` branch, and then `git merge develop` into this branch.

EwoutH commented:
I tried rebasing this branch to work further on it, but a lot of files where moved or renamed. So I will close this PR, and propose fixing the log statements (in d45db71, which maybe can be cherry picked) first, and then reformatting the strings to f-strings.
